lib/vyasa_web/live/mode_live/mediator.ex

```diff
@@ -36,22 +36,82 @@ defmodule VyasaWeb.ModeLive.Mediator do
   @impl true
 
   def handle_params(params, url, socket) do
+    url_params = params |> Map.put(:path, URI.parse(url).path)
+
     {:noreply,
      socket
-     |> assign(url_params: params |> Map.put(:path, URI.parse(url).path))
+     |> assign(url_params: url_params)
+     |> maybe_inject_mode()
      |> maybe_focus_binding()
-     |> sync_session()
-    }
+     |> sync_session()}
+  end
+
+  # injects mode if url slug contains mode, and there's an existing mode in socket state
+  defp maybe_inject_mode(
+         %{
+           assigns: %{
+             url_params: %{path: path},
```

Comment:
dont use url_params

its smarter to just take from handle params params since its clear we are pattern matching the router defined variables 

inject_mode can take params as argument and you wont have to wrangle with the socket on the pattern matching as much beyond getting the mode 

and you dont need this arbitrary 
```
 injected_mode =
      path
      |> String.split("/")
      |> Enum.at(1)
```

instead declare the :mode pattern match at router level